### PR TITLE
Added handling for single point presented as Vector

### PR DESCRIPTION
**Pull request by BioTurboNick**
Added Tree constructors that accept a Vector, convert it to a Matrix via `reshape`, and pass it along to the Matrix-oriented constructors. Also a couple formatting nits.

Fixes suggestion #76

### CODE REVIEW

**Comment by KristofferC** [comment]
`hcat` makes a copy so `reshape(a, 1, length(a))` is likely more efficient.

However, why cannot the `reshape` be done from the caller? It is a bit ambiguous whether `a = [1.0, 2.0]` is two points of dim 1 or one point of dim 2.

**Comment by BioTurboNick** [comment]
Thanks for pointing out the overhead of hcat vs. reshape; I had looked into reshape but didn't understand how to make it work for me. (fwiw, the right one would be `reshape(a, length(a), 1)`--I had assumed this command would just produce a 1d array again.)

For the latter, I think my suggestion works with the design of the interface for this package, which when given a 2d array, interprets each column as a point. And Julia base (if I'm using the term right) treats 1d arrays as columns in all other contexts.

If I create an array with multiple columns, Julia produces a 2d array natively:
```
julia> Array{Float64}([1 3 5])
1×3 Array{Float64,2}:
 1.0  3.0  5.0
```
This input is accepted by the Tree constructors as 3 1d points.

If I create an array with one column, Julia produces a 1d array natively:
```
julia> Array{Float64}([1; 3; 5])
3-element Array{Float64,1}:
 1.0
 3.0
 5.0
```

Thus, it should be expected by the user that passing 1d array, which is 1 column, to the tree constructor would be treated as a column == single point. Putting this reshaping function into the package wouldn't add much overhead and would remove the need for the caller to add unnecessary special casing code just to handle the condition of a single point.

**Comment by KristofferC** [comment]
> fwiw, the right one would be `reshape(a, length(a), 1)`

Hm, the dimension of the points is the number of rows so:

```
julia> a = rand(5); # 5, dim-1 points

julia> A = reshape(a, 1, length(a));

julia> dim, n_points = size(A)
(1, 5)
```

**Comment by BioTurboNick** [comment]
Yes, that's right. Maybe we're talking past each other? A single point is a 1-column, *d*-row array, which may be a *d*x1 Array{Float64,2} or a *d*-element Array{Float64,1}. Right?

I'm saying the latter type should be converted transparently to the former type. `knn` and `inrange` already do this for their inputs; the `NNTree` constructors do not.

**Comment by BioTurboNick** [comment]
I think the "codecov/patch" failed check is about not having tests for the new tree constructors, though I don't believe any of the tree constructors have tests? Unless I missed something.

**Comment by BioTurboNick** [comment]
I'd like to request review of this PR and possible integration.

**Comment by KristofferC** [comment]
This is to create a tree from one point? What's the point of that. Anyway, couldn't the original constructors just be extended to take a `VecOrMat`?

**Comment by KristofferC** [comment]
> The intention would be that the caller doesn't need to check whether they have just one point or multiple points before proceeding, creating cleaner code for basic usage.

But you still don't have to check that. If you have one point you would just have e.g. a `Matrix` with one column. Are there any real code where this check happens right now?

**Comment by BioTurboNick** [comment]
Just updated for compatibility with the current master.

**Comment by BioTurboNick** [comment]
Coming back to the performance bug comment: Yes possibly it would indicate a performance bug. But whether that matters to an end user is relative to their application, and it's something they could catch in profiling. Better to work slowly than to unexpectedly error out, I think?

**Comment by KristofferC** [comment]
This could use a test.

File: src/brute_tree.jl [hunk 1]
```diff
@@ -22,8 +22,8 @@ function BruteTree(data::AbstractVector{V}, metric::Metric = Euclidean();
     BruteTree(storedata ? data : Vector{V}(), metric, reorder)
 end
 
-function BruteTree(data::AbstractMatrix{T}, metric::Metric = Euclidean();
-                   reorder::Bool=false, leafsize::Int=0, storedata::Bool=true) where {T}
+function BruteTree(data::AbstractVecOrMat{T}, metric::Metric = Euclidean();
+                   reorder::Bool=false, leafsize::Int=0, storedata::Bool=true) where {T <: AbstractFloat}
```
Comment (KristofferC) [inline src/brute_tree.jl:26]: Is the new `T <: AbstractFloat` relevant for this PR?

Comment (BioTurboNick) [reply src/brute_tree.jl:26]: I believe I thought it was necessary to distinguish it from `BruteTree(data::AbstractVector{V}...) where {V <: AbstractVector}`, but I see that's probably not true. Both BallTree and KDTree currently provide that same type constraint. Doesn't look like it's necessary for BallTree or KDTree either.

**Comment by BioTurboNick** [comment]
Tests added!

**Pull request merged**